kernel/src/engine/arrow_expression.rs

```diff
@@ -191,20 +178,7 @@ fn evaluate_expression(
     use Expression::*;
     match (expression, result_type) {
         (Literal(scalar), _) => Ok(scalar.to_array(batch.num_rows())?),
-        (Column(name), _) => {
-            // TODO properly handle nested columns
-            // https://github.com/delta-incubator/delta-kernel-rs/issues/86
-            if name.contains('.') {
-                let mut path = name.split('.');
-                // Safety: we know that the first path step exists, because we checked for '.'
-                Ok(extract_column(batch, path.next().unwrap(), &mut path).cloned()?)
-            } else {
-                batch
-                    .column_by_name(name)
-                    .ok_or(Error::missing_column(name))
-                    .cloned()
-            }
-        }
+        (Column(name), _) => extract_column(batch, name.split('.')),
```

Comment:
do we still need to have a follow-up in order to properly handle nested columns (instead of just naive `split('.')`?)

Reply:
Definitely. My WIP nested column code already handles it.

Reply:
Ah, you refer to the deleted TODO comment. Reinstating it.